tcms/testruns/tests/test_permissions.py

```diff
@@ -128,3 +128,31 @@ def verify_post_with_permission(self):
             self.assertEqual(case.history.latest().history_id, case_run.case_text_version)
             self.assertEqual(last_run.build, case_run.build)
             self.assertEqual(None, case_run.close_date)
+
+
+class TestViewExecutionCommentsTestCase(PermissionsTestCase):
+    permission_label = 'django_comments.add_comment'
+    http_method_names = ['get']
+
+    @classmethod
+    def setUpTestData(cls):
+        cls.test_run = factories.TestRunFactory()
+
+        cls.url = reverse('testruns-get', args=[cls.test_run.pk])
+        cls.post_data={"summary": "mock"}
+        super().setUpTestData()
+
+        cls.add_comment_html = \
+            '<a href="#" class="addBlue9 js-show-commentdialog">{0}</a>' \
+            .format(_('Add'))
+
+    def assert_on_testrun_page(self):
+        response = self.client.get(self.url)
+        return response
+
+    def verify_get_with_permission(self):
+        self.assertContains(self.assert_on_testrun_page(), self.add_comment_html, html=True)
```

Comment:
Not like that. Your `assert_on_testrun_page` method needs to call assertContains independently and validate things like TestRun summary, product, list of test executions (maybe only summaries), etc.
You will call the method as any other method at the top of the `verify_` methods.